File: src/main/java/net/adventuregame/items/entities/BulletEntity.java
package net.adventuregame.items.entities;

import net.adventuregame.entities.Entity;
import net.adventuregame.models.TexturedModel;
import org.joml.Vector3f;

public class BulletEntity extends Entity {

    private Vector3f velocity;

    public BulletEntity(Vector3f position, Vector3f direction) {
        super(null, position, 0, 0, 0, 0.1f);
        this.velocity = new Vector3f(direction).normalize().mul(0.5f);
    }

    public void update() {
        increasePosition(velocity.x, velocity.y, velocity.z);
    }
}

File: src/main/kotlin/net/adventuregame/items/entities/BulletEntity.kt
package net.adventuregame.items.entities

import net.adventuregame.entity.Entity
import net.adventuregame.models.TexturedModel
import org.joml.Vector3f

class BulletEntity(position: Vector3f, direction: Vector3f) : Entity(position, 0f, 0f, 0f, 0.1f) {
    override fun GetModels(): List<TexturedModel> {
        return emptyList()
    }

    private val velocity: Vector3f

    init {
        this.velocity = Vector3f(direction).normalize().mul(0.5f)
    }

    fun update() {
        increasePosition(velocity.x, velocity.y, velocity.z)
    }
}
